Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp

```diff
@@ -125,4 +128,37 @@ Vector<String> transcodeImages(const Vector<String>& paths, const String& destin
     });
 }
 
+std::pair<String, Vector<IntSize>> utiAndAvailableSizesFromImageData(std::span<const uint8_t> data)
+{
+    Ref buffer = FragmentedSharedBuffer::create(data);
+    Ref imageDecoder = ImageDecoderCG::create(buffer.get(), AlphaOption::Premultiplied, GammaAndColorProfileOption::Applied);
+    imageDecoder->setData(buffer.get(), true);
+    auto uti = imageDecoder->uti();
+    if (!isSupportedImageType(uti))
+        return { { }, { } };
+
+    size_t frameCount = imageDecoder->frameCount();
+    // No support for animated image.
+    if (imageDecoder->repetitionCount() != RepetitionCountNone && frameCount > 1)
+        return { WTFMove(uti), { } };
+
+    Vector<IntSize> sizes;
+    sizes.reserveInitialCapacity(frameCount);
+    for (size_t index = 0; index < frameCount; ++index)
+        sizes.append(imageDecoder->frameSizeAtIndex(index));
+
+    return { WTFMove(uti), WTFMove(sizes) };
+}
+
+bool mayBeSVGData(const Settings& settings, std::span<const uint8_t> data)
+{
+    auto document = Document::create(settings, aboutBlankURL());
+    auto domParser = DOMParser::create(document.get());
```

Comment:
This is another layering violation. `Document` and `DOMParser` should not be used in the platform directory. I would suggest moving this function to SVGImage.cpp. It can be even a static function in `SVGImage` class. It can be named `SVGImage::isDataDecodable()` or `SVGImage::canDataBeDecoded()` or something. But this is not necessary if you want to keep it a standalone function.